modules/client/src/client-common/modules/graphql.ts

```diff
@@ -4,40 +4,55 @@ import { Context } from "../../client-common/context";
 import { QueryStatus } from "../graphql-queries";
 import { IClientGraphQLCore } from "../interfaces/core";
 
-export class GraphqlModule implements IClientGraphQLCore {
-  private _graphql: GraphQLClient[] = [];
-  private _graphqlIdx: number = -1;
+const GraphqlMap = new Map<GraphqlModule, GraphQLClient[]>();
+const GraphqlIdxMap = new Map<GraphqlModule, number>();
 
+export class GraphqlModule implements IClientGraphQLCore {
   constructor(context: Context) {
     if (context.graphql?.length) {
-      this._graphql = context.graphql;
-      this._graphqlIdx = Math.floor(Random.getFloat() * context.graphql.length);
+      GraphqlMap.set(this, context.graphql);
```

Comment:
```suggestion
      // Storing client data in the private module's scope to prevent external mutation
      GraphqlMap.set(this, context.graphql);
```